keyvault/shim.go

```diff
@@ -0,0 +1,158 @@
+package keyvault
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/pkcs12"
+)
+
+const (
+	pemContentType    = "application/x-pem-file"
+	pkcs12ContentType = "application/x-pkcs12"
+)
+
+type secretFetcher interface {
+	GetSecret(ctx context.Context, secretName string, opts *azsecrets.GetSecretOptions) (azsecrets.GetSecretResponse, error)
+}
+
+// Shim provides convenience methods for working with KeyVault.
+type Shim struct {
+	sf secretFetcher
+}
+
+// NewShim constructs a Shim for a KeyVault instance located at the provided url. The azcore.TokenCredential will
+// only be used during method calls, it is not verified at initialization.
+func NewShim(vaultURL string, cred azcore.TokenCredential) (*Shim, error) {
+	c, err := azsecrets.NewClient(vaultURL, cred, nil)
```

Comment:
If you make `sf` public, you can eliminate this constructor and the dep on `azcore`. It might also let you test with the `keyvault_test` package as well since you'd no longer need access to private members.

Reply:
I think a `New` is appropriate here; my benchmark is: if I incompletely initialize this struct will things break, that seems true here.